### ISSUE TITLE
Trying to use Oauth access_token to call v1/ APIs

### ISSUE THREAD

**Issue by tonydiaz** [square/connect-nodejs-sdk#69]
I get an error when I obtain a token from a merchant then call a `retrieveBusiness` and `listLocations` APIs to get their business information. Does the Oauth access token allow me to call v1 API? I'm doing this against the sandbox and I see the call going to production in the error (`https://squareup.com/v1/me`)

```js
const {
  OAuthApi,
  V1LocationsApi
} = require("square-connect");
const oauthApi = new OAuthApi();
const v1Locations = new V1LocationsApi();

const defaultClient = SquareConnect.ApiClient.instance;
defaultClient.basePath = "https://connect.squareupsandbox.com"

let oauth2 = defaultClient.authentications["oauth2"];
oauth2.accessToken = <my_access_token>


exports.squareAuthorize = functions.https.onRequest(
  async (request, response) => {
    var body = new SquareConnect.ObtainTokenRequest();
    body.client_id = <application_id>;
    body.client_secret = <application_secret>;
    body.grant_type = "authorization_code";
    body.code = request.query.code; //query param "code" sent by the request from square

    try {
      let authInfo = {};
      await oauthApi.obtainToken(body).then(data => {
        authInfo = data;
        console.log(
          "API called successfully. Returned data: " + JSON.stringify(data)
        );

        return data;
      });

       // access token from obtainToken
       oauth2.accessToken = authInfo.access_token;

      await v1Locations.retrieveBusiness().then(data => {
        console.log("API called successfully. Returned business data: " + data);
        return data;
      });
   } catch (e) {
       console.log(
          `[Error] Status:${e.status}, Messages: ${JSON.stringify(
            JSON.parse(e.response.text).errors,
            null,
            2
          )}`
        );
   } 


```

Error:
```
>  e : {"status":404,"response":{"req":{"method":"GET","url":"https://squareup.com/v1/me","data":null,"headers":{"user-agent":"Square-Connect-Javascript/2.20190925.0","accept-encoding":"gzip, deflate","square-version":"2019-09-25","accept":"application/json"}},"header":{"date":"Sun, 06 Oct 2019 17:32:38 GMT","content-type":"application/json; charset=utf-8","vary":"Accept-Encoding","cache-control":"no-cache","set-cookie":["_savt=92a0ba37-bb2e-453b-9217-5d2f7fb2XXXX; domain=.squareup.com; path=/; expires=Thu, 06 Oct 2022 17:32:38 -0000; secure"],"x-request-id":"cce7b44b-0e43-441f-9cb8-f6f2381cXXXX","x-frame-options":"sameorigin","x-content-type-options":"nosniff","x-xss-protection":"1; mode=block","x-download-options":"noopen","x-permitted-cross-domain-policies":"none","content-encoding":"gzip","content-length":"47","x-envoy-upstream-service-time":"17","strict-transport-security":"max-age=604800; includeSubDomains","server":"envoy","connection":"close"},"status":404,"text":"{\"message\":\"NotFoundError\"}"}}
>  [Error] Status:404, Messages: undefined
```

Another question around this topic is I want to be able to have a user connect their square account to my application but after I obtain the token there isn't any information I can use to tie the token to their customer profile in the application. Can I use the `state` query param in the Oauth to pass a customer ID from my application? https://developer.squareup.com/reference/square/oauth-api/authorize#response__property-state

**Comment by StephenJosey** [comment]
First off, I notice you mentioned you're OAuthing in sandbox, but doing the call in production...that is not going to work. Production does not know about sandbox tokens and vice-versa. You need to do the calls in sandbox:
```
var client = SquareConnect.ApiClient.instance;
// Set sandbox url
client.basePath = 'https://connect.squareupsandbox.com';
```
I think I'm misunderstanding your last question. OAuth is used to OAuth merchants (not customers) to use your application. Merchants can then have customers, but they are not customers themselves in the sense that they would have a `customer_id`. `ObtainToken` will return `merchant_id`, which is how you know who you're OAuthing. What exactly are you trying to accomplish?

**Comment by tonydiaz** [comment]
Sorry I didn't add the code but I point the basePath URL to the sandbox, somehow on `v1Locations.retrieveBusiness` call it points to `https://connect.squareup.com`

```js
const defaultClient = SquareConnect.ApiClient.instance;
defaultClient.basePath = "https://connect.squareupsandbox.com"
```

I also added it to the code above. 

For my last question, I think I figured it out. When a merchant uses Oauth on my app I can pass in an ID from my application through the `state` param and then tie the merchant ID and access token to their profile within my application.
